compiler/rustc_session/src/filesearch.rs

```diff
@@ -168,6 +168,11 @@ fn current_dll_path() -> Result<PathBuf, String> {
     Ok(OsString::from_wide(&filename).into())
 }
 
+#[cfg(all(target_os = "wasi", target_env = "p2"))]
```

Comment:
```suggestion
#[cfg(target_os = "wasi")]
```
why limit it to p2? this seems applicable to all wasi